modules/olm-catalogsource-image-template.adoc

```diff
@@ -0,0 +1,87 @@
+:global_ns: openshift-marketplace
+endif::[]
+
+[id="olm-catalogsource-image-template_{context}"]
+= Image template for custom catalog sources
+
+Operator compatibility with the underlying cluster can be expressed by a catalog source in various ways. One way, which is used for the default Red Hat-provided catalog sources, is to identify image tags for index images that are specifically created for a particular platform release, for example {product-title} 4.9.
+
+During a cluster upgrade, the index image tag for the default Red Hat-provided catalog sources are updated automatically by the Cluster Version Operator (CVO) so that Operator Lifecycle Manager (OLM) pulls the updated version of the catalog. For example during an upgrade from {product-title} 4.8 to 4.9, the `spec.image` field in the `CatalogSource` object for the `redhat-operators` catalog is updated from:
+
+[source,terminal]
+----
+registry.redhat.io/redhat/redhat-operator-index:v4.8
+----
+
+to:
+
+[source,terminal]
+----
+registry.redhat.io/redhat/redhat-operator-index:v4.9
+----
+
+However, the CVO does not automatically update image tags for custom catalogs. To ensure users are left with a compatible and supported Operator installation after a cluster upgrade, custom catalogs should also be kept updated to reference an updated index image.
+
+Starting in {product-title} 4.9, cluster administrators can add the `olm.catalogImageTemplate` annotation in the `CatalogSource` object for custom catalogs to an image reference that includes a template. The following Kubernetes version variables are supported for use in the template:
+
+* `kube_major_version`
+* `kube_minor_version`
+* `kube_patch_version`
+
+[NOTE]
+====
+You must specify the Kubernetes cluster version and not an {product-title} cluster version, as the latter is not currently available for templating.
+====
+
+Provided that you have created and pushed an index image with a tag specifying the updated Kubernetes version, setting this annotation enables the index image versions in custom catalogs to be automatically changed after a cluster upgrade. The annotation value is used to overwrite any image reference in the `spec.image` field of the `CatalogSource` object, whether or not the `spec.image` field was set. This helps avoid cluster upgrades leaving Operator installations in unsupported states or without a continued update path.
+
+[IMPORTANT]
+====
+You must ensure that the index image with the updated tag, in whichever registry it is stored in, is accessible by the cluster at the time of the cluster upgrade.
+====
+
+.Example catalog source with an image template
+[%collapsible]
+====
+[source,yaml,subs="attributes+"]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  generation: 1
+  name: example-catalog
+  namespace: openshift-marketplace
+  annotations:
+    olm.catalogImageTemplate:
+      "quay.io/example-org/example-catalog:v{kube_major_version}.{kube_minor_version}"
```

Comment:
This template can be `quay.io/example-v{kube_major_version}/example-catalog:v{kube_major_version}.{kube_minor_version}.{kube_patch_version}`

Reply:
I've updated this example here in the "Image template for custom catalog sources" section:

https://github.com/openshift/openshift-docs/compare/489079c504df85e18f798fd75131f80bc9b292b0..13ed417348a6db26e611b086999e333bb9d8aab1

See below for other instance.